### PR TITLE
feat: show fortnight and lab info in ClassCard

### PR DESCRIPTION
**Pull request by chinggg**
- use Regex to check `单|双` and `实验|上机` in `getPeriods()`
- add properties `fortnight: "单周" | "双周" | null` and `lab: "实验" | "上机" | null` in course object
- add `<div>` element in `ClassCard.vue`

image

It seems that `NoPeriodClassCard` is not likely to have the two properties, so I only made these changes.

### CODE REVIEW

File: src/mixins/ScheduleTable.js [hunk 1]
```diff
@@ -70,6 +70,8 @@ export const ScheduleTableMixin = {
             span: period[3],
             color: getColor(this.$store.state.previewClass.courseName, 0),
             isPreview: true,
+            fortnight: period[4] ? period[4] + "周" : null,
```
Comment (ZKLlab) [inline src/mixins/ScheduleTable.js:73]:
```suggestion
            fortnight: period[4] ? period[4] + '周' : null,
```
style: single quotes

File: src/mixins/ScheduleTable.js [hunk 2]
```diff
@@ -70,6 +70,8 @@ export const ScheduleTableMixin = {
...
+            lab: period[5]
```
Comment (ZKLlab) [inline src/mixins/ScheduleTable.js:74]:
```suggestion
            lab: period[5],
```
style: trailing comma

File: src/pages/index/components/ScheduleTable/ClassCard.vue [hunk 3]
```diff
@@ -7,6 +7,9 @@
     <div class="venue" v-else>
       <span class="venue-at">@</span>{{ $store.getters.extra(`${course.courseId}-${course.teacherId}`).venue }}
     </div>
+    <div class="extra" v-if="course.fortnight || course.lab">
+      {{ course.fortnight }} {{course.lab}}
```
Comment (ZKLlab) [inline src/pages/index/components/ScheduleTable/ClassCard.vue:11]:
```suggestion
      {{ course.fortnight }} {{ course.lab }}
```
style: add spaces in template interpolation

File: src/store.js [hunk 4]
```diff
@@ -53,6 +53,8 @@ export default new Vuex.Store({
               span: period[3],
               color: state.selectedClasses[courseId].themeColor,
               isPreview: false,
+              fortnight: period[4] ? period[4] + "周" : null,
```
Comment (ZKLlab) [inline src/store.js:56]:
```suggestion
              fortnight: period[4] ? period[4] + '周' : null,
```
style: single quotes

File: src/store.js [hunk 5]
```diff
@@ -53,6 +53,8 @@ export default new Vuex.Store({
...
+              lab: period[5]
```
Comment (ZKLlab) [inline src/store.js:57]:
```suggestion
              lab: period[5],
```
style: trailing comma

**Review by ZKLlab — CHANGES_REQUESTED** [review]
1. 移动端课程卡片组件也需要相应修改 src/pages/m/components/ScheduleTable/ClassCard.vue
2. 课程卡片新增的信息是否可以和教师/地点信息的样式区分开，使展示效果更加清晰？

**Comment by chinggg** [comment]
@ZKLlab 你认为新增信息应采用何种样式比较清晰？因为卡片颜色种类很多，所以我只好选择白色以免造成同种颜色覆盖

**Comment by ZKLlab** [comment]
@chinggg 先保持当前样式吧

**Review by ZKLlab — APPROVED** [review]

**Comment by chinggg** [comment]
好，我再 push 一下移动端的相应修改，然后 squash 合了，样式可以让新人来改

**Pull request merged**